File: src/javascript/util/fsa/OpenFilePickerOptionsType.java
package javascript.util.fsa;

import def.js.Array;
import def.js.Object;

/**
 * The types in the options used when opening a file
 *
 * @author gianpiero.diblasi
 */
public class OpenFilePickerOptionsType {

  public String description;
  public Object accept = new Object();

  public void pushAccept(String mime, Array<String> extensions) {
    this.accept.$set(mime, extensions);
  }
}

File: transpile/javascript/util/fsa/OpenFilePickerOptionsType.js
/**
 * The types in the options used when opening a file
 *
 * @author gianpiero.diblasi
 */
class OpenFilePickerOptionsType {

   description = null;

   accept = new Object();

   pushAccept(mime, extensions) {
    this.accept[mime] = extensions;
  }
}
